format/partition-statistics-spec.md

```diff
@@ -0,0 +1,43 @@
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+# Partition Statistics file format
+
+This is a specification for partition statistics files. It is designed to store statistics information 
+for every partition value as a row in the **table default format** sorted based on the `partition`.
+
+The schema of the file is as follows:
+
+| Field Id | Field Name              | Field Type   | isOptional | Doc                                                                                                                         |
+|----------|-------------------------|--------------|------------|-----------------------------------------------------------------------------------------------------------------------------|
+| 1        | partition               | StructType   | false      | See [PartitionData](https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/PartitionData.java) |                                                                                  
+| 2        | spec_id                 | IntegerType  | false      | Partition spec id                                                                                                           |
+| 3        | data_record_count       | LongType     | false      | Count of records in data files                                                                                              |                                               
+| 4        | data_file_count         | IntegerType  | false      | Count of data files                                                                                                         |
+| 5        | pos_delete_record_count | LongType     | true       | Count of records in position delete files                                                                                   |                                                          
+| 6        | pos_delete_file_count   | IntegerType  | true       | Count of position delete files                                                                                              |                                   
+| 7        | eq_delete_record_count  | LongType     | true       | Count of records in equality delete files                                                                                   |                                  
+| 8        | eq_delete_file_count    | IntegerType  | true       | Count of equality delete files                                                                                              |                                
```

Comment:
I'm wondering if we should also have a "last_modified" timestamp here. I think occasionally this comes up

Reply:
https://github.com/apache/iceberg/pull/7105#discussion_r1136556742

Reply:
I really wanted to handle these in phase 2 as mentioned in the design doc, if there is a popular demand, I can add it in this PR after #7581 is merged.

Reply:
Added `last_updated_ms` and `last_updated_snapshot_id` similar to what is there in partitions metadata table